docs/conditions.rst

```diff
@@ -48,6 +48,56 @@ The data type is of great importance for the rules engine. The following types a
 * floats (dot notation and scientific notation)
 * null
 
+Navigate structured data
+************************
+
+You can navigate strutured event, fact, var data objects using either dot notation or bracket notation:
+
+    .. code-block:: yaml
+
+      rules:
+        - name: Using dot notation
+          condition: event.something.nested == true
+          action:
+            debug:
+        - name: Analogous, but using bracket notation
+          condition: event.something["nested"] == true
+          action:
+            debug:
+
+Both of the above examples checks for the same value (attribute "nested" inside of "something") to be equal to `true`.
```

Comment:
@tarilabs What are the invalid charachters now, can we use [ ] in key names. Previously we had anything other than alphanumeric and underscore is invalid. Would we allow non english characters e.g.
```
---
- name: Simple Test
  hosts: all
  execution_strategy: parallel
  default_events_ttl: 46 day
  sources:
    - ansible.eda.generic:
        loop_count: 1
        timestamp: true
        create_index: i
        payload:
          - level: warning
            人: 的
  rules:
    - name: "Print the event"
      condition: event.人 is search("的", ignorecase=true) 
      action:
        print_event:
          pretty: True 
```